crates/bevy_math/src/compass.rs

```diff
@@ -0,0 +1,403 @@
+use crate::Dir2;
+#[cfg(feature = "bevy_reflect")]
+use bevy_reflect::Reflect;
+#[cfg(all(feature = "serialize", feature = "bevy_reflect"))]
+use bevy_reflect::{ReflectDeserialize, ReflectSerialize};
+
+/// A compass enum with 4 directions.
+#[derive(Debug, Clone, Copy, PartialEq, Eq, Hash)]
+#[cfg_attr(feature = "serialize", derive(serde::Serialize, serde::Deserialize))]
+#[cfg_attr(feature = "bevy_reflect", derive(Reflect), reflect(Debug, PartialEq))]
+#[cfg_attr(
+    all(feature = "serialize", feature = "bevy_reflect"),
+    reflect(Deserialize, Serialize)
+)]
+pub enum CompassQuadrant {
+    /// The north direction.
+    North,
+    /// The east direction.
+    East,
+    /// The south direction.
+    South,
+    /// The west direction.
+    West,
+}
+
+/// A compass enum with 8 directions.
+#[derive(Debug, Clone, Copy, PartialEq, Eq, Hash)]
+#[cfg_attr(feature = "serialize", derive(serde::Serialize, serde::Deserialize))]
+#[cfg_attr(feature = "bevy_reflect", derive(Reflect), reflect(Debug, PartialEq))]
+#[cfg_attr(
+    all(feature = "serialize", feature = "bevy_reflect"),
+    reflect(Deserialize, Serialize)
+)]
+pub enum CompassOctant {
+    /// The north direction.
+    North,
+    /// The north east direction.
+    NorthEast,
+    /// The east direction.
+    East,
+    /// The south east direction.
+    SouthEast,
+    /// The south direction.
+    South,
+    /// The south west direction.
+    SouthWest,
+    /// The west direction.
+    West,
+    /// The north west direction.
+    NorthWest,
+}
+
+impl From<CompassQuadrant> for Dir2 {
+    fn from(q: CompassQuadrant) -> Self {
```

Comment:
```suggestion
    /// [`CompassOctantant::North`] corresponds to [`Dir2::Y`].
    fn from(q: CompassQuadrant) -> Self {
```

Reply:
And then similar comments for all 4 `From` impls :) Not blocking, but I think it's useful.